bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/ConfigStore.java

```diff
@@ -354,7 +354,7 @@ public String mapItemUIDtoHueID(Item item) {
     public String getHueUniqueId(final String hueId) {
         String unique = hueId;
         try {
-            unique = String.format("%02X", Integer.valueOf(hueId));
+            unique = String.format("%X", Integer.valueOf(hueId));
```

Comment:
Can you provide a bit more detail as to what this is fixing and what doesn't work with the current unique id?  Last time I looked at the Hue API it states the format is AA:BB:CC:DD:EE:FF:00:11-XX.

Have you tested this with Alexa? The last change to the unique id was to fix Alexa discovery which seems to be very temperamental when it comes to the unique id

Reply:
Sure: Philips TVs are omitting leading zeros when using the Ambilight-Hue Feature, so the current approach has ID 03 for example and the TV interprets it as 3 and that obviously doesn't work. As the Philips TV obviously works with a real Philips Hue something in this implementation is not correct. Maybe we should start at 10 with the numbers instead if they have to be 2 digits?

Reply:
Also the official documentation doesn't mention any leading zeros: https://developers.meethue.com/develop/get-started-2/

They mention ID 1 and not ID 01 like we are currently doing it.

Reply:
The code you've changed is for setting the uniqueid field.

If you've spotted the I'd problem on 3.0 and are referring to the id used in something like `/api/<username>/lights/<id>/state` this should already be fixed in 3.1 by https://github.com/openhab/openhab-addons/pull/9463
--

Reply:
Alright, I'll try reverting this one and see what happens then. This PR would only be for fixing the response to xy change requests.

Reply:
I updated this PR accordingly